tools-ui/src/main/java/com/hedera/hashgraph/client/ui/utilities/KeyPairUtility.java

```diff
@@ -73,18 +89,61 @@ private KeyPair getKeyPair(File pemFile, String message) {
 				Arrays.fill(pwd, 'x');
 			} catch (HederaClientException e) {
 				pwd = askForPasswordAgain(pemFile);
-				if (pwd == null) {
+				if (pwd == null || Arrays.equals(new char[0], pwd)) {
 					return null;
 				}
 			}
 		}
 		return keyPair;
 	}
 
+	/**
+	 * Given a pemFile, resets the password and returns it
+	 *
+	 * @param pemFile
+	 * 		the file where the pem file is stored;
+	 * @return the new password
+	 */
+	public static char[] resetPassword(String pemFile) throws HederaClientException, KeyStoreException {
+		var properties = new UserAccessibleProperties(DEFAULT_STORAGE + File.separator + USER_PROPERTIES, "");
+
+		// Check Hashcode
+		var storedHashCode = properties.getMnemonicHashCode();
+		var hashCode = Ed25519KeyStore.getMnemonicHashCode(pemFile);
+		if (hashCode != null && hashCode != storedHashCode) {
+			logger.info("The key is not associated with the current mnemonic");
+			PopupMessage.display("Error recovering password",
+					"The key is not associated with the current recovery phrase. The password cannot be changed",
+					"CONTINUE");
+			return new char[0];
+		}
+
+		// get password bytes
+		var token = properties.getHash();
+		var decoder = Base64.getDecoder();
+		var index = Ed25519KeyStore.getIndex(pemFile);
+		var tokenBytes = decoder.decode(token);
+		if (tokenBytes.length < Constants.SALT_LENGTH + KEY_LENGTH / 8) {
+			logger.error("Token size check failed");
```

Comment:
If this happens, should the process continue or quit here? Seems to me it should quit immediately.

Reply:
good catch. will change